File: python-selenium/pages/car_detail_page.py
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.ui import WebDriverWait

from .locators import CarDetailPageLocators
from .page import BasePage

class CarDetailPage(BasePage):
    def __init__(self, web_driver):
        super().__init__(web_driver)

        wait = WebDriverWait(self.driver, 5)
        wait.until(expected_conditions.presence_of_element_located(CarDetailPageLocators.CAR_DETAILS))

        self.trim_levels = self.driver.find_element(*CarDetailPageLocators.TRIM_LEVELS)
        self.year = self.driver.find_element(*CarDetailPageLocators.YEAR)
        self.make = self.driver.find_element(*CarDetailPageLocators.MAKE)
        self.model = self.driver.find_element(*CarDetailPageLocators.MODEL)
        self.edit_car_link = self.driver.find_element(*CarDetailPageLocators.EDIT_CAR_LINK)
        self.delete_car_button = self.driver.find_element(*CarDetailPageLocators.DELETE_CAR_BUTTON)

    def get_edit_car_link(self):
        return self.edit_car_link

    def click_edit_car_link(self):
        self.edit_car_link.click()

    def click_delete_car_button(self):
        self.delete_car_button.click()

File: java-car-data/car-data-web-ui-tests/src/test/java/com/glicerial/samples/cardata/web/uitests/page/CarDetailPage.java
package com.glicerial.samples.cardata.web.uitests.page;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;


public class CarDetailPage {

    private WebDriver driver;

    @FindBy(id = "editlink")
    private WebElement editCarLink;

    @FindBy(id = "deletecarbutton")
    private WebElement deleteCarButton;

    @FindBy(id = "year")
    private WebElement year;

    @FindBy(id = "make")
    private WebElement make;

    @FindBy(id = "model")
    private WebElement model;

    @FindBy(id = "trimlevels")
    private WebElement trimLevels;

    public CarDetailPage(WebDriver webDriver) {
        this.driver = webDriver;
        WebDriverWait wait = new WebDriverWait(driver, 5);
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("trimlevels")));
        PageFactory.initElements(driver, this);
    }

    public WebElement getYear() {
        return year;
    }

    public WebElement getMake() {
        return make;
    }

    public WebElement getModel() {
        return model;
    }

    public WebElement getTrimLevels() {
        return trimLevels;
    }

    public WebElement getEditCarLink() {
        return editCarLink;
    }

    public void clickEditCarLink() {
        editCarLink.click();
    }

    public void clickDeleteCarButton() {
        deleteCarButton.click();

        WebDriverWait wait = new WebDriverWait(driver, 5);

        // Confirm delete
        wait.until(ExpectedConditions.alertIsPresent());
        driver.switchTo().alert().accept();

        // Dismiss dialog that says "Deleted."
        wait.until(ExpectedConditions.alertIsPresent());
        driver.switchTo().alert().accept();

        // Wait for javascript redirect
        wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//div[@class='maincontent']//table[@id='carstable']")));
    }
}
